src/main/java/dukeApp/files/TaskList.java

```diff
@@ -0,0 +1,148 @@
+package dukeApp.files;
+import dukeApp.action.Action;
+import java.util.ArrayList;
+
+public class TaskList {
+    public ArrayList<Task> aList;
+
+    public TaskList(ArrayList<Task> aList) {
+        this.aList = aList;
+    }
+
+    public TaskList() {}
+
+    public int getSize() {
+        return aList.size();
+    }
+
+    /**
+     * Prints task in the list
+     */
+    public void printList() {
+        Task t;
+        System.out.println("Here are the tasks in your list:");
+        for (int i=1; i <= aList.size(); i++) {
+            t = aList.get(i-1);
+            System.out.println(i + ". [" +t.getType()+ "][" +t.getStatusIcon()+ "]" +t.toString());
+        }
+        System.out.println("");
+    }
+
+    /**
+     * Actions to be done for delete & done tasks
+     * @param rank task number in the list
+     * @param dd string indicating delete or done action
+     */
+    public void deleteDone(int rank, String dd) throws DukeException {
```

Comment:
Almost thought you were deleting the task once it was done haha, might want to change function name to something like checkIfDeleteOrDone